python/lsst/ts/observatory/control/remote_group.py

```diff
@@ -135,43 +232,52 @@ def __init__(self, components, domain=None, log=None, intended_usage=None):
         self.long_timeout = 30.0
         self.long_long_timeout = 120.0
 
-        self._components = components
-        self._component_names = []
+        self._components = dict(
+            [
+                (component, component.lower().replace(":", "_"))
+                for component in components
+            ]
+        )
 
         self.domain = domain if domain is not None else salobj.Domain()
 
-        _remotes = {}
+        self._usages = None
+
+        self.rem = types.SimpleNamespace()
 
-        for i, component in enumerate(self._components):
+        for component in self._components:
             name, index = salobj.name_to_name_index(component)
-            resources = self.get_required_resources(component, intended_usage)
-            rname = component.lower().replace(":", "_")
-            self._component_names.append(rname)
+            rname = self._components[component]
+            resources = self.get_required_resources(rname, intended_usage)
+
             if resources.add_this:
-                _remotes[rname] = salobj.Remote(
-                    domain=self.domain,
-                    name=name,
-                    index=index,
-                    readonly=resources.readonly,
-                    include=resources.include,
+                setattr(
+                    self.rem,
+                    rname,
+                    salobj.Remote(
+                        domain=self.domain,
+                        name=name,
+                        index=index,
+                        readonly=resources.readonly,
+                        include=resources.include,
+                    ),
                 )
             else:
-                _remotes[rname] = None
-
-        self.rem = types.SimpleNamespace(**_remotes)
+                setattr(self.rem, rname, None)
 
         self.scheduled_coro = []
 
+        # Mark components that where excluded from the resources to not be
+        # checked.
         self.check = types.SimpleNamespace(
-            **{component: True for component in self._component_names}
+            **{c: getattr(self.rem, c) is not None for c in self.components_attr}
         )
 
         self.start_task = asyncio.gather(
             *[
                 getattr(self.rem, c).start_task
+                for c in self.components_attr
```

Comment:
It might increase clarity to do the remote lookup loop once, e.g.:
```
# Dict of component attribute name: remote, if present, else None
attr_remotes = {attr: getattr(self.rem, attr) for attr in self.components_attr}

# Mark components that were excluded from...
self.check = types.SimpleNamespace(**{c: remote is not None for c, remote in attr_remotes.items()})

self.start_task = asyncio.gather(*[remote.start_task for remote in attr_remotes.items() if remote is not None])
```